### PR TITLE
feat: adding managent command to clear error state

### PR DESCRIPTION
**Pull request by kiram15**
**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- x ] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/openedx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/openedx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise/management/commands/update_config_last_errored_at.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        """
+        For each audit record kind (learner and content), find all the records in batch, and lookup
+        if they've had recent sync errors in the last day. If not, clear out the last_content_sync_errored_at
+        value associated with the configuration.
+        """
+        try:
+            has_learner_errors, has_content_errors = True, True
+            yesterday = datetime.now() - timedelta(days=1)
+            for channel_code, (ConfigModel, LearnerAuditModel) in MODELS.items():
+                configs = ConfigModel.objects.all().filter()
```
Comment (alex-sheehan-edx) [inline enterprise/management/commands/update_config_last_errored_at.py:78]: is the filter here necessary?

File: enterprise/management/commands/update_config_last_errored_at.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                        enterprise_customer=config.enterprise_customer,
+                        integrated_channel_code=channel_code,
+                        plugin_configuration_id=plugin_id,
+                        api_response_status_code__gt=299
+                    )
+                    if not errored_content_audits:
+                        config.last_content_sync_errored_at = None
+                        has_content_errors = False
+
+                    if not has_learner_errors and not has_content_errors:
```
Comment (alex-sheehan-edx) [inline enterprise/management/commands/update_config_last_errored_at.py:106]: why is `last_sync_errored_at` dependent on both no learner and content errors whereas the `last_learner_sync_errored_at` is only dependent on learner errors?

**Review by alex-sheehan-edx — COMMENTED** [review]
mostly good, just a few implementation q's!

File: tests/test_management.py [hunk 3]
```diff
@@ -1930,3 +1931,77 @@ def test_normal_run(self):
...
+        assert csod_config.last_sync_errored_at is None
+        assert csod_config.last_content_sync_errored_at is None
+        assert csod_config.last_learner_sync_errored_at is None
+
+    def test_invalid_audits(self):
+        """
+        Verify that the management command runs when all records have the same subdomain
+        """
+        old_timestamp = datetime.now() - timedelta(days=5)
+        print('before ', old_timestamp)
```
Comment (alex-sheehan-edx) [inline tests/test_management.py:1985]: cruft

File: tests/test_management.py [hunk 4]
```diff
@@ -1930,3 +1931,77 @@ def test_normal_run(self):
...
+        old_timestamp = old_timestamp.replace(tzinfo=pytz.UTC)
+        print('after ', old_timestamp)
```
Comment (alex-sheehan-edx) [inline tests/test_management.py:1987]: more cruft `:all-the-cruft:`

File: enterprise/management/commands/update_config_last_errored_at.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                        config.last_sync_errored_at = None
+                    config.save()
```
Comment (alex-sheehan-edx) [inline enterprise/management/commands/update_config_last_errored_at.py:108]: Do we want to log some context here?

**Review by alex-sheehan-edx — COMMENTED** [review]
sorry, couple more questions!

File: enterprise/management/commands/update_config_last_errored_at.py (hunk 2 above)
Comment (kiram15) [reply enterprise/management/commands/update_config_last_errored_at.py:106]: So there's 3: last, last_learner, and last_content. We use them all in different contexts, but as you said last sync is dependent on both content and learner to show the user whats the most recent error regardless of type.

File: enterprise/management/commands/update_config_last_errored_at.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+    def update_config_last_errored_at(self):
+        """
+        For each audit record kind (learner and content), find all the records in batch, and lookup
+        if they've had recent sync errors in the last day. If not, clear out the last_content_sync_errored_at
+        value associated with the configuration.
+        """
+        try:
+            has_learner_errors, has_content_errors = True, True
+            yesterday = datetime.now() - timedelta(days=1)
```
Comment (alex-sheehan-edx) [inline enterprise/management/commands/update_config_last_errored_at.py:76]: should this be UTC?

**Review by alex-sheehan-edx — APPROVED** [review]
One question about timezones, and as long as you address the linter, this LGTM! 😄

**Pull request merged**